internal/go/skycfg/proto_message.go

```diff
@@ -355,9 +356,79 @@ func maybeConvertString(v reflect.Value, t reflect.Type) (reflect.Value, bool) {
 	return v, true
 }
 
+// maybeConvertToWrapper checks if [sky] is a primitive and [t] is a corresponding
+// protobuf wrapper type and attempts to convert it.
+// Returns a non-nil Value and nil error if successfully converted, a non-nil error
+// if unsuccessful conversion, and nil Value and nil error if no conversion attempt
+// was required.
+func maybeConvertToWrapper(t reflect.Type, sky starlark.Value) (*reflect.Value, error) {
+	BoolValueType := reflect.TypeOf(&wrappers.BoolValue{})
+	StringValueType := reflect.TypeOf(&wrappers.StringValue{})
+	DoubleValueType := reflect.TypeOf(&wrappers.DoubleValue{})
+	Int32ValueType := reflect.TypeOf(&wrappers.Int32Value{})
+	Int64ValueType := reflect.TypeOf(&wrappers.Int64Value{})
+	BytesValueType := reflect.TypeOf(&wrappers.BytesValue{})
+
+	switch sky := sky.(type) {
+	case starlark.Bool:
+		if t == BoolValueType {
+			val := reflect.New(t.Elem())
+			val.Elem().Set(reflect.ValueOf(wrappers.BoolValue{Value: bool(sky)}))
+			return &val, nil
+		}
+	case starlark.Int:
+		switch t {
+		case Int64ValueType:
+			val := reflect.New(t.Elem())
+			int64val, _ := sky.Int64()
+			val.Elem().Set(reflect.ValueOf(wrappers.Int64Value{Value: int64val}))
+			return &val, nil
+		case DoubleValueType:
+			val := reflect.New(t.Elem())
+			int64val, _ := sky.Int64()
+			val.Elem().Set(reflect.ValueOf(wrappers.DoubleValue{Value: float64(int64val)}))
+			return &val, nil
+		case Int32ValueType:
+			val := reflect.New(t.Elem())
+			int64val, _ := sky.Int64()
+			if int64val >= math.MinInt32 && int64val <= math.MaxInt32 {
```

Comment:
I believe you can use `starlark.AsInt32(sky)` to handle this error case.

Reply:
yep!